### ISSUE TITLE
Make ApiRoutes discoverable for ApiExplorer

### ISSUE THREAD

**Issue by Xceno** [OrchardCMS/Orchard#7165]
While trying to integrate [Swagger](http://swagger.io/) with Orchard I dug into the [Swashbuckle](https://github.com/domaindrivendev/Swashbuckle) source. It uses the [ApiExplorer](https://msdn.microsoft.com/de-de/library/system.web.http.description.apiexplorer(v=vs.118).aspx) internally to extract all the route infos.

However, it seems the `ApiExplorer` is unable to find any routes in Orchard, therefore its `ApiDescriptions`-Property is always empty. I wanted to subclass it but all the interesting stuff is private...

I've also tried copying routes manually into the RouteTable and a lot of other things but to no avail.
There also an [old thread on CodePlex](http://orchard.codeplex.com/discussions/636606) dealing with this issue, but the proposed solution doesn't seem to work (anymore).
Sadly the documentation around the ApiExplorer is pretty scarce too.

**TL;DR**
Is there any way to get the ApiExplorer to discover Orchards WebApi Routes?

**Comment by sebastienros** [comment]
No, it would require a custom ApiExplorer that would get all the routes from the providers. And probably also have some extensibility point to support Aliases, or not as they are not APIs.

**Comment by SzymonSel** [comment]
Actually it didn't -> [https://github.com/OnefoursevenLabs/Orchard-Swagger](https://github.com/OnefoursevenLabs/Orchard-Swagger)
Done in a bit hacky fashion.

**Comment by Xceno** [comment]
@SzymonSel This is fantastac! Thank you for looking into this. Gonna check out your source ASAP; I wonder what I've missed back then.
